dao/OrderRepository.go

```diff
@@ -0,0 +1,118 @@
+package dao
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+)
+
+type OrderPostgres struct {
+	db *sql.DB
+}
+
+func NewOrderPostgres(db *sql.DB) *OrderPostgres {
+	return &OrderPostgres{db: db}
+}
+
+type Order struct {
+	IdDeliveryService int    `json:"delivery_service_id,omitempty"`
+	Id           int    `json:"id"`
+	IdCourier         int    `json:"courier_id,omitempty"`
+	DeliveryTime      string `json:"delivery_time,omitempty"`
+	CustomerAddress   string `json:"customer_address,omitempty"`
+	Status            string `json:"status"`
+	OrderDate         string `json:"order_date"`
+	RestaurantAddress string `json:"restaurant_address"`
+	Picked            bool   `json:"picked"`
+}
+
+type DetailedOrder struct {
+	IdDeliveryService  int    `json:"delivery_service_id,omitempty"`
+	Id            int    `json:"id"`
+	IdCourier          int    `json:"courier_id,omitempty"`
+	DeliveryTime       string `json:"delivery_time,omitempty"`
+	CustomerAddress    string `json:"customer_address,omitempty"`
+	Status             string `json:"status,omitempty"`
+	OrderDate          string `json:"order_date,omitempty"`
+	RestaurantAddress  string `json:"restaurant_address,omitempty"`
+	Picked             bool   `json:"picked,omitempty"`
+	CourierName        string `json:"name,omitempty"`
+	CourierPhoneNumber string `json:"phone_number,omitempty"`
+}
+
+func (r *OrderPostgres) AssigningOrderToCourier_InDB(order Order) error {
+	transaction, err := r.db.Begin()
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Println("connected to db")
+
+	defer transaction.Commit()
+	s := fmt.Sprintf("UPDATE delivery SET courier_id = %d WHERE id = %d", order.IdCourier, order.Id)
+	log.Println(s)
+	insert, err := transaction.Query(s)
+	defer insert.Close()
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	return nil
+}
+
+func(r *OrderPostgres) GetAllServiceCompletedOrders_fromDB(limit, page, idService int) ([]DetailedOrder, int){
+	var Orders []DetailedOrder
+	transaction, err := r.db.Begin()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer transaction.Commit()
+	res, err := transaction.Query(fmt.Sprintf("SELECT delivery.order_date, delivery.courier_id,delivery.id,delivery.customer_address,delivery.restaurant_address,couriers.name,couriers.phone_number FROM delivery JOIN couriers ON couriers.id_courier=delivery.courier_id Where delivery.status='completed' and delivery.delivery_service_id=%d LIMIT %d OFFSET %d", idService, limit, limit*(page-1)))
```

Comment:
https://github.com/Baraulia/AUTHORIZATION_SERVICE/pull/7#discussion_r812616458
you can give aliases to tables, it would make request easier